libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Recognizers/EntityRecognizers/ChannelMentionEntityRecognizer.cs

```diff
@@ -0,0 +1,133 @@
+    /// </summary>
+    /// <remarks>
+    /// This makes it easy to pass explicit mentions from channels like Teams/Skype to LUIS models.
+    /// </remarks>
+    public class ChannelMentionEntityRecognizer : Recognizer
+    {
+        /// <summary>
+        /// Class identifier.
+        /// </summary>
+        [JsonProperty("$kind")]
+        public const string Kind = "Microsoft.ChannelMentionEntityRecognizer";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChannelMentionEntityRecognizer"/> class.
+        /// </summary>
+        /// <param name="callerPath">Optional, source file full path.</param>
+        /// <param name="callerLine">Optional, line number in source file.</param>
+        [JsonConstructor]
+        public ChannelMentionEntityRecognizer([CallerFilePath] string callerPath = "", [CallerLineNumber] int callerLine = 0)
+            : base(callerPath, callerLine)
+        {
+        }
+
+        /// <inheritdoc/>
+        public override Task<RecognizerResult> RecognizeAsync(DialogContext dialogContext, Activity activity, CancellationToken cancellationToken = default, Dictionary<string, string> telemetryProperties = null, Dictionary<string, double> telemetryMetrics = null)
+        {
+            var result = new RecognizerResult();
+
+            // promote external mention entities from the activity into recognizer result
+            if (activity.Entities != null)
+            {
+                if (result.Entities == null)
+                {
+                    result.Entities = new JObject();
+                }
+
+                dynamic entities = result.Entities;
+
+                // convert activity.entities entity that looks like this:
+                // {
+                //    "type": "mention",
+                //    "mentioned": {
+                //        "id": "28:0047c760-1f42-4a78-b1bd-9ecd95ec3615",
+                //        "name": "Tess"
+                //    },
+                //    "text": "<at>Tess</at>"
+                // },
+                int iStart = 0;
+                foreach (dynamic entity in activity.Entities.Where(e => e.Type == "mention").Select(e => JObject.FromObject(e)))
+                {
+                    // into recognizeresult that looks like this:
+                    // "entities": {
+                    //   "mention": [
+                    //      "28:0047c760-1f42-4a78-b1bd-9ecd95ec3615"
+                    //   ],
+                    //   "$instance": {
+                    //     "mention": [
+                    //        {
+                    //            "startIndex": 10,
+                    //            "endIndex": 13,
+                    //            "score": 1.0,
+                    //            "text": "@tom",
+                    //            "type": "mention",
+                    //            "resolution": {
+                    //              "value": "@tom"
+                    //            }
```

Comment:
Why is this here?  If it is really a resolution belongs in entities.mention which should be a structure and otherwise isn't it just the text?